### PR TITLE
Blogit: - Feature/processing in models 🏭 🗄️  👨🏾‍💻 

### PR DESCRIPTION
**Pull request by Bennyjoez**
## Blogit: - Feature/processing in models 🏭 🗄️  👨🏾‍💻 🪶 

In this milestone, I managed to implement the following: 🪶 
- Create model classes for all entities as shown in the ERD diagram 🎨 
- Set up associations between models. 👪 
- Create at least one user by running the given code. 👨🏾‍💻 
- Create at least 4 posts written by one of the users you created by running the given code. 👨🏾‍💻 
- Create at least 6 posts comments for one of the posts you created by running the given code. 👨🏾‍💻 
- Use CRUD methods.
- Create custom methods on the models. ➕ 
   - A method that returns the 3 most recent posts for a given user. ➕ 
   - A method that updates the posts counter for a user. ➕ 
   - A method that returns the 5 most recent comments for a given post. ➕ 
   - A method that updates the comments counter for a post. ➕ 
   - A method that updates the likes counter for a post. ➕ 

💯

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+  def recent_comments
+    comments.order(created_at: :desc).limit(5)
+  end
+
+  private
+
+  def update_user_posts_counter
+    author.update_column(:posts_counter, author.posts_counter.to_i + 1)
+  end
```
Comment (julie-ify) [inline app/models/post.rb]: - [ ] **[OPTIONAL]** You can use `increment!` to auto-update the `posts_counter`

  ```
  def update_user_posts_counter
    author.increment!(:posts_counter)
  end
  ```

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :post, class_name: 'Post', foreign_key: 'post_id'
+
+  after_create :update_likes_counter
+
+  private
+
+  def update_likes_counter
+    post.update_column(:likes_counter, post.likes_counter.to_i + 1)
+  end
```
Comment (julie-ify) [inline app/models/like.rb]: - [ ] **[OPTIONAL]** You can use `increment!` to auto-update the `likes_counter`

  ```
  def update_likes_counter
    post.increment!(:likes_counter)
  end
  ```

File: app/models/comment.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :post, class_name: 'Post', foreign_key: 'post_id'
+
+  after_create :update_comments_counter
+
+  private
+
+  def update_comments_counter
+    post.update_column(:comments_counter, post.comments_counter.to_i + 1)
+  end
```
Comment (julie-ify) [inline app/models/comment.rb]: - [ ] **[OPTIONAL]** You can use `increment!` to auto-update the `comments_counter`

  ```
  def update_comments_counter
    post.increment!(:comments_counter)
  end
  ```

**Review by julie-ify — APPROVED** [review]
Hi @Bennyjoez,

## Highlights!
- Good code organization and associations ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Bennyjoez** [comment]
Thank you for the good suggestions @julie-ify 👍🏾 
I managed to effect the changes. 💯

**Pull request merged**